pkg/actor/partitioned_update.go

```diff
@@ -128,10 +129,14 @@ func (up *partitionedUpdate) Act(ctx context.Context, cluster *resource.Cluster,
 
 	// test to see if we are running inside of Kubernetes
 	// If we are running inside of k8s we will not find this file.
-	runningInsideK8s := inK8s("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	var (
+		runningInsideK8s = inK8s("/var/run/secrets/kubernetes.io/serviceaccount/token")
+		k8sNamespace     = readK8Sfile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	)
 
-	serviceName := cluster.PublicServiceName()
-	if runningInsideK8s {
+	var serviceName string
+	if runningInsideK8s && k8sNamespace == cluster.Namespace() {
```

Comment:
Instead of handling a special case where we happen to be running in the same namespace, let's just always connect to the namespaced or fully qualified service name?
```suggestion
        // We might need .svc.cluster.local as the suffix here?
        serviceName := fmt.Sprintf("%s.%s", cluster.PublicServiceName(), cluster.Namespace())
	if runningInsideK8s {
```

Reply:
the fully qualified service can be different from `svc.cluster.local` if u set custom local domain